addons/mrp/models/stock_orderpoint.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class StockWarehouseOrderpoint(models.Model):
+    _inherit = 'stock.warehouse.orderpoint'
+
+    show_bom = fields.Boolean('Show BoM column', compute='_compute_show_bom')
+    bom_id = fields.Many2one(
+        'mrp.bom', string='Bill of Materials',
+        domain="[('type', '=', 'normal'), '|', ('product_id', '=', product_id), '&', ('product_id', '=', False), ('product_tmpl_id', '=', product_tmpl_id)]")
```

Comment:
the orderpoint has a required company_id, you should only be able to select bom within the same company or without company